Deprecate v2 api [1001]

MarkZuber opened the pull request:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1001

MarkZuber commented:
This is ready to functionally review.  Most of the building blocks are in place and our code is now again warning free from obsolete markers.

I had to [Ignore] a couple of unit tests specifically around mockability of the IPublicClientApplication until we get that API issue resolved so that we are mockable and that's still in progress/prototyping.

All of the MigrationAid classes I've moved are [Obsolete("TODO: document")] since I want to get this functionally correct and then we can focus on the aka.ms links and appropriate documentation for the various cases.  I would love some assistance there to get this right.

So please review this for functionality and ensuring we're deprecating the right APIs and that the code is in the right place.  Then we'll do a follow up PR before we release that updates the comments and documentation appropriately.

henrik-me commented:
> using System;

 not sure I understand the impact of some of the changes in this file. Will follow up in-person

---
Refers to: src/Microsoft.Identity.Client/ClientApplicationBase.cs:28 in d031dbd. [](commit_id = d031dbd90557e6881b6ade73d1698c48fc84c134, deletion_comment = False)

henrik-me commented on src/Microsoft.Identity.Client/Platforms/Android/UIParent.cs:
```diff
@@ -83,9 +88,12 @@ public UIParent(Activity activity, bool useEmbeddedWebview) : this(activity)
-        : this(ValidateParentObject(parent), useEmbeddedWebview)
+            : this(ValidateParentObject(parent), useEmbeddedWebview)
         {
+            throw new NotImplementedException("See https://aka.ms/msal-net-3-breaking-changes");
```
>throw new NotImplementedException("See https://aka.ms/msal-net-3-breaking-changes"); [](start = 12, length = 84)

Thought, Make a factory method somewhere which does this?

henrik-me commented on src/Microsoft.Identity.Client/Platforms/Android/UIParent.cs:
```diff
@@ -49,7 +49,8 @@ public sealed class UIParent
         /// Default constructor. Should not be used on Android.
         /// </summary>
-        [Obsolete("This constructor should not be used because this object requires a parameters of type Activity. ")]
+        [Obsolete("See https://aka.ms/msal-net-3-breaking-changes", true)]
```
>Obsolete [](start = 9, length = 8)

assuming all the obsolete stuff moved to the migrationaid subfolder (may require #ifdef and marking the class partial)

MarkZuber replied:
Not for UIParent.  I'm marking it obsolete inline since it's platform-specific and there are multiple implementations of the same class.  so i'm just updating the obsolete documentation for these and throwing exception like we're doing for the other stuff that is moving to migrationaid.cs

---
In reply to: [268374935](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1002#discussion_r268374935) [](ancestors = 268374935)

henrik-me commented:
>     public sealed class UIParent

any way of removing UIParent completely?

---
Refers to: src/Microsoft.Identity.Client/Platforms/Android/UIParent.cs:44 in d031dbd. [](commit_id = d031dbd90557e6881b6ade73d1698c48fc84c134, deletion_comment = False)

MarkZuber commented:
>     public sealed class UIParent

I was going to propose that we do since all we need are the static IsSystemWebViewAvailable method(s) in the various implementations.  Due to platform specifics, we'll need to keep these here to obsolete them since there are 5-6 different implementations of this class for each platform.  but we've talked about how/where to move IsSystemWebViewAvailable to, and I think we should completely obliterate this class and move that functionality somewhere else.

---
In reply to: [475826013](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1002#issuecomment-475826013) [](ancestors = 475826013)

---
Refers to: src/Microsoft.Identity.Client/Platforms/Android/UIParent.cs:44 in d031dbd. [](commit_id = d031dbd90557e6881b6ade73d1698c48fc84c134, deletion_comment = False)

MarkZuber commented:
> using System;

1.  we moved out obsolete code to MigrationAid.cs
2.  I moved the .WithBuilders.cs code into here and removed .WithBuilders.cs so the diff looks scarier than it should be.

---
In reply to: [475825538](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1002#issuecomment-475825538) [](ancestors = 475825538)

---
Refers to: src/Microsoft.Identity.Client/ClientApplicationBase.cs:28 in d031dbd. [](commit_id = d031dbd90557e6881b6ade73d1698c48fc84c134, deletion_comment = False)

henrik-me commented:
>     public sealed class UIParent

Most of these are the same across platforms. I'm ok to leave it here but think it would also be possible to move it. I agree that oblibrating would be the way to go. Thought we should do it.... :) Perhaps a new issue

---
In reply to: [475826730](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1002#issuecomment-475826730) [](ancestors = 475826730,475826013)

---
Refers to: src/Microsoft.Identity.Client/Platforms/Android/UIParent.cs:44 in d031dbd. [](commit_id = d031dbd90557e6881b6ade73d1698c48fc84c134, deletion_comment = False)

MarkZuber commented:
>     public sealed class UIParent

[https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1005](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1005)

---
In reply to: [475827190](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1002#issuecomment-475827190) [](ancestors = 475827190,475826730,475826013)

---
Refers to: src/Microsoft.Identity.Client/Platforms/Android/UIParent.cs:44 in d031dbd. [](commit_id = d031dbd90557e6881b6ade73d1698c48fc84c134, deletion_comment = False)

henrik-me commented on src/Microsoft.Identity.Client/PublicClientApplication.cs:
```diff
@@ -69,434 +54,55 @@ namespace Microsoft.Identity.Client
         {
 #if NET_CORE
             throw new PlatformNotSupportedException("On .NET Core, interactive authentication is not supported. " + 
                 "Consider using Device Code Flow https://aka.ms/msal-net-device-code-flow or Integrated Windows Auth https://aka.ms/msal-net-iwa");
```
>Consider using Device Code Flow [](start = 17, length = 31)

Should we suggest alto that they look at bringing their own browser or using the system browser approach (though we don't have it implemented yet, but we will soon)?

henrik-me commented on tests/Microsoft.Identity.Test.SideBySide/UnifiedCacheTests.cs:
```diff
@@ -35,6 +35,8 @@
 using msal::Microsoft.Identity.Client;
 using Microsoft.Identity.Test.LabInfrastructure;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using msal::Microsoft.Identity.Client.AppConfig;
```
>msal [](start = 6, length = 4)

interesting. Was this supposed to move to a M.I.C namespace instead of it's own namespace?

MarkZuber replied:
There's a separate issue for moving out AppConfig/ApiConfig namespaces which will be in a separate PR.

henrik-me commented:
Generally I don't see any issues with this PR. We will have to do more stuff with the public API, but assuming that is all in other work items.
